File: mobile/lib/domain/entities/lesson_lab_item.dart
import 'package:equatable/equatable.dart';

class LessonLabItemEntity extends Equatable {
  final int id;
  final int lessonId;
  final String category; // equipment | reagent | element | vessel
  final String name;
  final String? formula;
  final String? quantity;
  final String? unit;
  final String? notes;
  final int sortOrder;
  final bool isRequired;
  final bool isActive;

  const LessonLabItemEntity({
    required this.id,
    required this.lessonId,
    required this.category,
    required this.name,
    this.formula,
    this.quantity,
    this.unit,
    this.notes,
    required this.sortOrder,
    required this.isRequired,
    required this.isActive,
  });

  @override
  List<Object?> get props => [
        id,
        lessonId,
        category,
        name,
        formula,
        quantity,
        unit,
        notes,
        sortOrder,
        isRequired,
        isActive,
      ];
}

File: backend/app/Models/LessonLabItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class LessonLabItem extends Model
{
    public const CATEGORY_EQUIPMENT = 'equipment';
    public const CATEGORY_REAGENT = 'reagent';
    public const CATEGORY_ELEMENT = 'element';
    public const CATEGORY_VESSEL = 'vessel';

    protected $fillable = [
        'lesson_id',
        'category',
        'name',
        'formula',
        'quantity',
        'unit',
        'notes',
        'sort_order',
        'is_required',
        'is_active',
    ];

    protected $casts = [
        'sort_order' => 'integer',
        'is_required' => 'boolean',
        'is_active' => 'boolean',
    ];

    public function lesson(): BelongsTo
    {
        return $this->belongsTo(Lesson::class);
    }
}
